service/history/task/cross_cluster_task_processor.go

```diff
@@ -303,6 +314,10 @@ func (p *crossClusterTaskProcessor) processTaskRequests(
 		}
 		cancel()
 
+		if p.ctx.Err() != nil {
+			return
+		}
```

Comment:
would this be better above, in the loop?  or is that unsafe to interrupt?  (though if anything under it checks the context, it may still be interrupted)

Reply:
Sure. I don't have a strong opinion on this. Moving it into the for loop also works and probably can save us some iterations during the shutdown. 

I was thinking the for loop is already non-blocking once the context is cancelled on shutdown, and we just need one check before sending the response back instead of checking it in each iteration.